### PR TITLE
File 200

### PR DESCRIPTION
**Pull request by hannesstockner**

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/transfer/Service.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  case class SoftDeleteServiceError(message: String) extends SoftDeleteError
+  case object SoftDeleteEnvelopeNotFound extends SoftDeleteError
+  case object SoftDeleteEnvelopeAlreadyDeleted extends SoftDeleteError
+  case object SoftDeleteEnvelopeInWrongState extends SoftDeleteError
+
+  def softDelete(delete: (EnvelopeId) => Future[Boolean], getEnvelope: (EnvelopeId) => Future[FindResult])
+                (envelopeId: EnvelopeId)
+                (implicit ex: ExecutionContext): Future[SoftDeleteResult] =
+    delete(envelopeId).flatMap {
+      case true => Future { Xor.Right(envelopeId) }
```
Comment (konradwudkowski) [inline app/uk/gov/hmrc/fileupload/transfer/Service.scala]: Future.successful will be better as there is no need for a separate thread here

File: test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  val failed = Future.failed(new Exception("not good"))
+
+  def newController(softDelete: EnvelopeId => Future[SoftDeleteResult] = _ => failed) =
+    new TransferController(softDelete)
+
+
+  "Delete envelope" should {
+    "return response with 200" in {
+      val envelope = Support.envelope
+      val request = FakeRequest("DELETE", s"/envelopes/${envelope._id}")
```
Comment (konradwudkowski) [inline test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala]: I'd keep it as FakeRequest() only as adding "DELETE" and url suggests that controllers needs it but it doesn't. These information is used by the router but here we bypass routing and pass a request directly to the controller.

File: test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      val result = controller.nonStubDelete(envelope._id)(request).futureValue
+
+      result.header.status shouldBe Status.OK
+    }
+
+    "return response with 400" in {
+      val envelope = Support.envelope
+      val request = FakeRequest("DELETE", s"/envelopes/${envelope._id}")
+
+      val controller = newController(softDelete = _ => Future.successful(Xor.Left(SoftDeleteServiceError("not good"))))
```
Comment (konradwudkowski) [inline test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala]: Seeing this error type I expected 500 error and not 400. The convention in other services was that FooServiceError was an unexpected error that translates to a 500 in the controller, here it translates to a BadRequest. Maybe we could call it SoftDeleteBadRequest or sth explicitly representing a bad request?

File: app/uk/gov/hmrc/fileupload/transfer/Service.scala (hunk 1 above)
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/transfer/Service.scala]: +1

File: test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala (hunk 3 above)
Comment (hannesstockner) [inline test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala]: +1

File: test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala (hunk 2 above)
Comment (hannesstockner) [inline test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala]: +1

**Pull request merged**